azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/participant/grid/cell/ParticipantGridCellAvatarView.kt

```diff
@@ -116,16 +143,12 @@ internal class ParticipantGridCellAvatarView(
     private fun setDisplayName(displayName: String) {
         avatarView.name = displayName
         avatarView.invalidate()
-
-        if (displayName.isBlank()) {
-            displayNameAudioTextView.visibility = GONE
-        } else {
-            displayNameAudioTextView.text = displayName
-        }
+        setTextViewDisplayName(displayName)
     }
 
     private fun setMicButtonVisibility(isMicButtonVisible: Boolean) {
-        if (!isMicButtonVisible) {
+        val status = participantViewModel.getParticipantStatusStateFlow().value
```

Comment:
No such logic should be in the view. `isMicButtonVisible: Boolean` should already take in consideration the participant status in the view model and be unit tested.